File: src/api/autostart.ts
import { invoke } from '@tauri-apps/api/core'

/**
 * 检查是否启用自动启动
 */
export async function isAutostartEnabled(): Promise<boolean> {
  return await invoke('is_autostart_enabled')
}

/**
 * 启用自动启动
 */
export async function enableAutostart(): Promise<void> {
  return await invoke('enable_autostart')
}

/**
 * 禁用自动启动
 */
export async function disableAutostart(): Promise<void> {
  return await invoke('disable_autostart')
}

File: src-tauri/src/commands/autostart.rs
use tauri_plugin_autostart::ManagerExt;

#[tauri::command]
pub async fn is_autostart_enabled(app: tauri::AppHandle) -> Result<bool, String> {
    app.autolaunch()
        .is_enabled()
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn enable_autostart(app: tauri::AppHandle) -> Result<(), String> {
    app.autolaunch()
        .enable()
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn disable_autostart(app: tauri::AppHandle) -> Result<(), String> {
    app.autolaunch()
        .disable()
        .map_err(|e| e.to_string())
}
